metadata-ingestion/src/datahub/ingestion/source/unity/proxy_profiling.py

```diff
@@ -152,21 +161,47 @@ def _check_analyze_table_statement_status(
                 statement_id
             )
             self._raise_if_error(response, "get-statement")
-            status = response.status
+            status = response.status  # type: ignore
 
         return status.state == StatementState.SUCCEEDED
 
     def _get_table_profile(
         self, ref: TableReference, include_columns: bool
     ) -> TableProfile:
-        table_info = self._workspace_client.tables.get(ref.qualified_table_name)
+        table_info = self._get_table_info(ref, include_columns)
         return self._create_table_profile(table_info, include_columns=include_columns)
 
+    def _get_table_info(
+        self, ref: TableReference, include_columns: bool
+    ) -> Union[TableInfo, Table]:
+        table_info: Union[TableInfo, Table]
+        if self.hive_metastore_proxy and ref.catalog == HIVE_METASTORE:
+            table_info = self.hive_metastore_proxy._get_table(
+                Schema(
+                    id=ref.schema,
+                    name=ref.schema,
+                    # This is okay, as none of this is used in profiling
+                    catalog=self.hive_metastore_proxy.hive_metastore_catalog(None),
+                    comment=None,
+                    owner=None,
+                ),
+                ref.table,
+                include_column_stats=include_columns,
+            )
+        else:
+            table_info = self._workspace_client.tables.get(ref.qualified_table_name)
+        return table_info
```

Comment:
```suggestion
        if self.hive_metastore_proxy and ref.catalog == HIVE_METASTORE:
            return self.hive_metastore_proxy._get_table(
                Schema(
                    id=ref.schema,
                    name=ref.schema,
                    # This is okay, as none of this is used in profiling
                    catalog=self.hive_metastore_proxy.hive_metastore_catalog(None),
                    comment=None,
                    owner=None,
                ),
                ref.table,
                include_column_stats=include_columns,
            )
        else:
            return self._workspace_client.tables.get(ref.qualified_table_name)
```

Reply:
updated this. separated hive metastore profiles generation logic into hive_metastore_proxy module.